# Summarizing and Computing Descriptives Statistics

In [6]:
# example

import numpy as np
import pandas as pd

df = pd.DataFrame([[1.4,np.nan],[7.1,-4.5],
                  [np.nan,np.nan],[0.75,-1.3]],
                  index = ['a','b','c','d'],
                  columns=['one','two']
                 )
df

one  two
a  1.40  NaN
b  7.10 -4.5
c   NaN  NaN
d  0.75 -1.3

In [7]:
# sum return the series containing column sums

df.sum()

one    9.25
two   -5.80
dtype: float64

In [8]:
# if we want the sum of columns then we pass the axis=column

df.sum(axis='columns')

a    1.40
b    2.60
c    0.00
d   -0.55
dtype: float64

- __if we use skipna = False then it will skip the column whih have na value__
- __skipna skip the missing value, and by default it is true__

In [12]:


df.mean(axis="columns", skipna=False)

a      NaN
b    1.300
c      NaN
d   -0.275
dtype: float64

 - __function 'idxmin' and 'idxmax' return statistics like the index value where the minimum and maximum values are attained__

In [14]:
# Example
# in 'one' column maximum value exist in 'b' and same goes with 'two' column

df.idxmax()

one    b
two    d
dtype: object

In [17]:
# in 'one' column minimum value exist in 'd' and same goes with 'two' column

df.idxmin()

one    d
two    b
dtype: object

- __'cumsum' function add first row values with second row and put that in second row__
- __for example  1.40 + 7.10 = 8.50 and put that in b row where missing value did not change anything__
- __at 'd' row 8.50 + 0.75 = 9.25, -4.5 + (-1.3) = -5.8__

In [20]:
df.cumsum()

one  two
a  1.40  NaN
b  8.50 -4.5
c   NaN  NaN
d  9.25 -5.8

- __describe function__

In [21]:
# describe function describe it

df.describe()

one       two
count  3.000000  2.000000
mean   3.083333 -2.900000
std    3.493685  2.262742
min    0.750000 -4.500000
25%    1.075000 -3.700000
50%    1.400000 -2.900000
75%    4.250000 -2.100000
max    7.100000 -1.300000

### Correlation and Covariance
- __summary statistics like correlation, covarianve are computed from the pairs of arguement__

In [ ]:
# To find the percentage 

a = price.pct_change()
a

In [ ]:
# To find the correlation
b = corr(some argument)
b

In [ ]:
# To find the covariance
c = cov(some argument)
c

### Unique values, vlaue count, and membership


In [24]:
# Example to find unique values

obj = pd.Series(['c','a','d','a','a','b','b','c','c'])
obj.unique()

array(['c', 'a', 'd', 'b'], dtype=object)

In [26]:
# Cout values

obj.value_counts()

a    3
c    3
b    2
d    1
dtype: int64

In [28]:
#Sorting 

obj.value_counts(sort=True)

a    3
c    3
b    2
d    1
dtype: int64

In [31]:
# is in

obj.isin(['a','d'])

0    False
1     True
2     True
3     True
4     True
5    False
6    False
7    False
8    False
dtype: bool